### ISSUE TITLE
Troubles using updateDataElement

### ISSUE THREAD

**Issue by mijnschoen** [pitchmuc/launchpy#12]
First off all: thanks for the wonderfull work on this module. It helps me a lot getting information on our setup much quicker.
But now I'm trying to update a data elements settings and keep failing.

I want to update the settings of a data element by extending a list inside of it.
What I try:

1. I've selected the DataElement (in `lookup_de`) from the list returned by `my_property.getDataElements()`
2. I've also captured the attributes using `lookup_de_attr = lp.extractAttributes(lookup_de)`
3. As I kept failing in updating, I now just tried directly putting this inside the updateDataElement method: `result = my_property.updateDataElements(lookup_de['id'], lookup_de_attr)`

But I keep getting an error as result:
```
{'id': 'b73fac27-d04a-46ee-9763-7b4a366c0fb1', 'status': '400', 'code': 'schema-validation-error', 'title': 'Schema Validation Error', 'detail': 'The request did not contain the following required properties: data', 'meta': {}, 'source': {}}
```

Any idea what I'm doing wrong?

Launchpi version ==0.3.3
Requests version ==2.25.1

**Comment by mijnschoen** [comment]
Hi @pitchmuc,
I think the issue has to do with the way the patch method is invoked. 
The `obj` object is created with the data property, but this is send along as the `params` argument, where I would expect this to be the `data` argument.

I tested changing the call to the self.connector.patchData method (within the updateDataElements method) by using the following, which seems to work:
```python
dataElements = self.connector.patchData(
            #config.endpoints['global']+'/data_elements/'+dataElement_id, obj, header=self.header)
            config.endpoints['global'] + '/data_elements/' + dataElement_id, data=obj, header=self.header)
```

**Comment by pitchmuc** [comment]
Hello @mijnschoen 
Thanks for raising this issue. 
You are correct and happy that you have found a way around it. 
When changing architecture, I made the different requests methods more generic but forgot to change my call inside the absracted methods. 
So something like this:
`self.connector.patchData(self.endpoint+path, obj, header=self.header)` 
should be 
`self.connector.patchData(self.endpoint+path, data=obj, header=self.header)` 

I will try to publish a patch today for this. 
I found several instances of that issue.
